src/components/VisualizationPlugin/VisualizationPluginWrapper.js

```diff
@@ -10,6 +10,7 @@ const VisualizationPluginWrapper = (props) => {
     const [pluginProps, setPluginProps] = useState(props)
     const [isLoading, setIsLoading] = useState(true)
     const [error, setError] = useState(null)
+    const { onResponsesReceived = Function.prototype } = props
```

Comment:
I actually intentionally did not use `defaultProps` because I read that it is deprecated. Though there is an argument for using it to keep it consistent in the codebase.

https://github.com/jsx-eslint/eslint-plugin-react/issues/2396

https://react.dev/learn/passing-props-to-a-component#specifying-a-default-value-for-a-prop

Since I can't props in the function arguments, this seemed like the next best solution to avoid deprecated ways of doing things. Weirdly, when I tried just having `props.onResponseReceived` as a dependency for the `useEffect`, the linter complained that it needed the whole props object

![image](https://github.com/user-attachments/assets/ff96ca77-7758-4974-b8f5-2a5e93892c01)

Reply:
I think the current implementation is fine. If we did things as in your last suggestion we would effectively be removing the `onResponsesReceived` from the `props` object. This in turn would mean that everywhere we now pass the entire `props` object we'd have to start passing `{ ...props, onResponsesReceived}`.